sandbox/libs/authn/docs/jwt.md

```diff
@@ -0,0 +1,43 @@
+# JWTs in OpenSearch
+
+JWTs are one of the current methods OpenSearch uses to authenticate Rest requests to an OpenSearch cluster.
+
+When the `jwt_auth_domain` is configured in the `authc` section of the security configuration file, the security plugin will verify the JWT with the `signing_key` configured and decrypt the payload to extract `subject` and `roles` information from the claims in the payload of the JWT.
+
+JWTs are mainly generated in the following 3 ways:
+
+1) Generated after first authentication with a SAML server and used in subsequent requests to authenticate the user to prevent rounttrips to the SAML identity provider
+    - For SAML authentication, the security plugin takes the SAML response and extracts `claims` from the response to embed in the JWT. The `exchange_key` is the key used to sign the token. The key is created by the security plugin.
+    - A separate `jwt_auth_domain` does not need to be configured when `saml_auth_domain` is in the list of authentication backends
+2) After authentication with an OIDC IdP the security plugin gets a JWT from the JSON web key set (JWKS) endpoint of the IdP
+    - Keys or shared secrets do not need to be configured in security `config.yml` because these retrieved from the IdP.
+3) JWT Auth domain - Using a shared signing key - A separate authentication server that can authenticate and create JWTs is compatible with the JSON Auth domain for the security plugin. The payload of the jwt needs to minimally contain a subject (keyed by `subject_key`) and roles (keyed by `roles_key`)
+
+
+These JWTs are utilized by using an HTTP Authorization header like so:
+
+`curl -XGET -H "Authorization: Bearer ${ACCESS_TOKEN}" http://localhost:9200`
+
+This is often referred to as Bearer Authentication
+
+
+# Tokens for Delegated Authority
+
+The following is a proposal for how to best secure asynchronous tasks running in OpenSearch or OpenSearch extensions.
+
+When a task or job is run in OpenSearch, the executor of the task should be explicitly defined. By explicitly specifying the user context for a job, it makes
+it clearer to understand what privileges are in use when the job is executing and gives an administrator the ability to fine tune the security model for their cluster.
+In addition to offering better fine tuning for security, it also gives administrators the power to revoke privileges to prevent an errant job from executing.
+
+In order to delegate authority for a background task or job, I propose to create a token vending service to produce tokens that confer access to the cluster and
+let's the background task assume a subject when executing.
+
+There are 2 types of tokens:
+
+- Refresh Token - A Refresh Token is longer lasting and used to receive short-lived Access Tokens that actually confer access
+- Access Token - Access Tokens are short-lived tokens that confer access
```

Comment:
What kind of access granting model are we attaching?  would raw permissions be in the access token, or would it only have the principal for lookup within the local node for permissions?

Reply:
Initial thoughts is that the token does not contain any claim regarding roles and a mechanism needs to be created to resolve the subject to a list of permissions.

Reply:
```suggestion
- Access Token - Access Tokens are short-lived tokens that confer trusted authentication information, no claims are included
```

Small tweak to the language, what do you think?

Reply:
The tokens may contain claims, but it will not contain authorization information